ui/src/validation.ts

```diff
@@ -0,0 +1,62 @@
+import validatejs from "validate.js"
+import { derived, writable, Readable, get } from "svelte/store"
+
+export enum ValidationStatus {
+  NotStarted = "NOT_STARTED",
+  FormatValidation = "FORMAT_VALIDATION",
+  Pending = "PENDING",
+  Error = "ERROR",
+  Success = "SUCCESS"
+}
+
+type ValidationState =
+  { status: ValidationStatus.NotStarted } |
+  { status: ValidationStatus.FormatValidation; input: string } |
```

Comment:
Is this ever constructed?

Reply:
Nope, it's something I was playing with as a state in the state machine but I don't think it's actually necessary